### PR TITLE
Friends

### LINKED ISSUE #13: FS: Friends
### Front-end:

- A front-end web page to add friends (see latest mockup for details)
- Sketch goal (filter by name for following and suggested friends)
### Backend:
- Friends model - a user follows many friends
-  API routes:
   - get suggested friends (random) - search query (search by name)
   -  get your following - search query by name
   -  follow / unfollow friend (POST request)

### Integration
- Suggested actually fetches users from the backend
- I can send a friend request
- I can undo a sent friend requests 
- I can accept a received friend request
- I can cancel a received friend request
- I can remove a friend

### PR DESCRIPTION
**Pull request by yoniakabecky**
close #13 

### Front-end:
 - [x] A front-end web page to add friends (see latest mockup for details)
 - [x] Sketch goal (filter by name for following and suggested friends)
### Backend:
 - [x] Friends model - a user follows many friends
 -  API routes:
     -  suggested friends
        - [x] get suggested friends (random)
        - [x] search query (search by name)
     -  following
        - [x] get your following
        - [x] search query by name
     - [x] follow / unfollow friend (POST request)
### Integration:
 - [x] Suggested actually fetches users from the backend
 - [x] I can send a friend request
 - [x] I can undo a sent friend requests
 - [x] I can accept a received friend request
 - [x] I can cancel a received friend request
 - [x] I can remove a friend

### CODE REVIEW

**Review by bonnieli — APPROVED** [review]
great attention to details!

File: client/src/components/FriendsPage/FriendsPage.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    maxWidth: 500,
+    height: "100%",
+  },
+})(MuiPaper);
+
+export default function FriendsPage() {
+  const [selectedTab, setSelectedTab] = React.useState(0);
+  const [loading, setLoading] = React.useState(true);
+  const [fetchedFriends, setFetchedFriends] = React.useState(
+    new Array(categories.length).fill([])
```
Comment (sina-jamshidi) [inline client/src/components/FriendsPage/FriendsPage.js]: It's not a huge deal but are these default values necessary? It seems like fetchedFriends (and maybe keywords) can just be defaulted to `[]` and still work.

File: server/controllers/friendsControllers.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        name: {
+          $regex: req.query.search,
+          $options: "i",
+        },
+      };
+    }
+
+    const suggestions = await User.find(query)
+      .sort({ createdAt: -1 })
+      .limit(15);
```
Comment (sina-jamshidi) [inline server/controllers/friendsControllers.js:45]: lol nice

**Review by sina-jamshidi — APPROVED** [review]
Just one small note. Looks good to me!

File: client/src/components/FriendsPage/FriendsPage.js (hunk 1 above)
Comment (yoniakabecky) [reply client/src/components/FriendsPage/FriendsPage.js]: Ah, that's right. I'm gonna change that👍 
(Might not change keywords... )

**Pull request merged**